src/hamster-service.py

```diff
@@ -38,7 +40,7 @@
     quit()
 
 
-class Storage(db.Storage, dbus.service.Object):
+class Storage(db.Storage, dbus.service.Object, DbusSessionListener):
```

Comment:
I wonder if adding a mixin class here is the cleanest way to implement this. Also, it doesn't seem a responsibility of a class called Storage to implement a "stop-tracking-on-shutdown", but maybe that class already does more than its name implies (haven't looked too closely now).

Reply:
I agree with that. It's probably the simplest way but not the cleanest one.

I think I can create a separate object having a reference on the storage object to call StopTracking. To make it even more separated, this object can send a stop tracking message on the dbus, it could be even a separate process.